x-pack/plugins/actions/server/builtin_action_types/lib/request_oauth_token.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import qs from 'query-string';
+import axios from 'axios';
+import stringify from 'json-stable-stringify';
+import { Logger } from '../../../../../../src/core/server';
+import { request } from './axios_utils';
+import { ActionsConfigurationUtilities } from '../../actions_config';
+import { RewriteResponseCase } from '../../../../actions/common';
+
+export interface OAuthTokenResponse {
+  tokenType: string;
+  accessToken: string;
+  expiresIn: number;
+}
+
+export async function requestOAuthToken<T>(
+  tokenUrl: string,
+  params: T,
+  grantType: string,
+  configurationUtilities: ActionsConfigurationUtilities,
+  logger: Logger,
+  rewriteBodyRequest: RewriteResponseCase<T>
+): Promise<OAuthTokenResponse> {
+  const axiosInstance = axios.create();
+
+  const res = await request({
+    axios: axiosInstance,
+    url: tokenUrl,
+    method: 'post',
+    logger,
+    data: qs.stringify({
+      ...rewriteBodyRequest(params),
```

Comment:
nit: Would it make more sense for `requestOAuthJWTToken` and `requestOAuthClientCredentialsToken` to call `rewriteBodyRequest` and pass in the params already formatted and ready to use?